hw02-ArrayList/src/main/java/DIYArrayList.java

```diff
@@ -0,0 +1,198 @@
+import java.util.*;
+import java.util.function.UnaryOperator;
+
+public class DIYArrayList<T> implements List<T> {
+    private static final Object[] EMPTY_ITEM = {};
+    private Object[] array;
+    private int curSize;
+
+    public DIYArrayList(int size) {
+        if (size >= 0) {
+            array = new Object[size];
+            this.curSize = 0;
+        }
+        else {
+            throw new IllegalArgumentException("Invalid DIYArrayList size: " + size);
+        }
+    }
```

Comment:
Лучше сначала проверить условия (аргументы), а потом выполнять всю работу.
С одним аргументом разница может не сильно заметна, но когда аргументов больше, то так удобнее писать.

```suggestion
    public DIYArrayList(int size) {
        if (size < 0)
            throw new IllegalArgumentException("Invalid DIYArrayList size: " + size);
        
        array = new Object[size];
        this.curSize = 0;
    }
```